crypto/init.c

```diff
@@ -646,6 +665,11 @@ int OPENSSL_init_crypto(uint64_t opts, const OPENSSL_INIT_SETTINGS *settings)
         return 0;
 #endif
 
+#ifndef OPENSSL_NO_CMP
```

Comment:
I would be tempted to move this into the CMP logging code so that it happens immediately before you write to the log. As it currently stands the cmp log gets inited every time - even if you're not doing anything related to CMP.

Reply:
I've re-structured the logging facility such that there is no need to call `OSSL_CMP_log_open()`, at least not at lib init.